### PR TITLE
chore: add mongoose and change project structure

### PR DESCRIPTION
**Pull request by fjsandov**

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
-  PORT: process.env.PORT || 3000
+  PORT: process.env.PORT || 3000,
+  DATABASE_URL: 'mongodb://localhost/notify-watcher',
```
Comment (negebauer) [inline config.js]:
```suggestion
  DATABASE_URL: process.env. DATABASE_URL || 'mongodb://localhost/notify-watcher',
```

Comment (negebauer) [inline config.js]:
```suggestion
  DATABASE_URL: process.env.DATABASE_URL || 'mongodb://localhost/notify-watcher',
```

File: index.js [hunk 2]
```diff
@@ -1,37 +1,4 @@
...
-routers.forEach(router => {
-  app.use(router.routes());
-  app.use(router.allowedMethods());
-});
-
-app.use(logger());
-
-const server = app.listen(config.PORT);
-
-async function startup() {
```
Comment (negebauer) [inline index.js]: The startup function for watchers should be here (it's not part of the Koa app)
And as [per this comment](https://github.com/notify-watcher/server/pull/11#discussion_r336170625) from @sasalatart, `app.listen` should be called _after_ watchers are downloaded and running

```js
async function startup() {
   mongoose.connect(config.DATABASE_URL);

   const { watchers } = await setUpWatchers();
   config.WATCHERS = watchers;

   app.listen(config.PORT);
}

startup()
```

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const Koa = require('koa');
+const logger = require('koa-logger');
+const mongoose = require('mongoose');
+const config = require('../config');
+const setUpWatchers = require('./watchers');
+const router = require('./routes');
+
+mongoose.connect(config.DATABASE_URL);
```
Comment (negebauer) [inline src/app.js]: move to main `index.js`

File: src/app.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const app = new Koa();
+
+app.use(router.routes());
+app.use(router.allowedMethods());
+
+app.use(logger());
+
+async function startup() {
```
Comment (negebauer) [inline src/app.js]: move to main `index.js`

File: index.js (hunk 2 above)
Comment (sasalatart) [reply index.js]: +1

Although I think that `mongoose.connect` is `async`, so we should probably wait for it before going on with the rest of the execution.

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+const router = require('./routes');
+
+mongoose.connect(config.DATABASE_URL);
+
+const app = new Koa();
+
+app.use(router.routes());
+app.use(router.allowedMethods());
+
+app.use(logger());
```
Comment (sasalatart) [inline src/app.js]: Is this working ok? I think it should be placed before the routes.

File: index.js (hunk 2 above)
Comment (negebauer) [reply index.js]: yeah, edited the comment

File: src/app.js (hunk 3 above)
Comment (fjsandov) [reply src/app.js]: But some endpoints use the mongoose queries.

File: config.js [hunk 6]
```diff
@@ -2,9 +2,27 @@ const path = require('path');
...
-  PORT: process.env.PORT || 3000,
-  DOWNLOAD_WATCHERS: process.env.DOWNLOAD_WATCHERS || false,
+  env: {
+    isDev,
+    isTest,
+    isProd,
+  },
+  PORT,
+  DATABASE_URL: databaseUrl(),
+  DOWNLOAD_WATCHERS: DOWNLOAD_WATCHERS || false,
```
Comment (negebauer) [inline config.js]:
```suggestion
  DOWNLOAD_WATCHERS: DOWNLOAD_WATCHERS,
```

File: config.js [hunk 7]
```diff
@@ -2,9 +2,27 @@ const path = require('path');
 
 const TEMP_DIR_PATH = path.resolve(path.join('.', 'tmp'));
 
+const { NODE_ENV, DATABASE_URL, PORT = 3000, DOWNLOAD_WATCHERS } = process.env;
```
Comment (negebauer) [inline config.js]:
```suggestion
const { NODE_ENV, DATABASE_URL, PORT = 3000, DOWNLOAD_WATCHERS = false } = process.env;
```

File: config.js [hunk 8]
```diff
@@ -2,9 +2,27 @@ const path = require('path');
```
Comment (negebauer) [inline config.js]: this file should be `src/config.js`

File: index.js [hunk 9]
```diff
@@ -1,37 +1,12 @@
-const Koa = require('koa');
```
Comment (negebauer) [inline index.js]: this file should be `src/index.js`

**Pull request merged**